cpp/src/gandiva/exported_funcs.cc

```diff
@@ -0,0 +1,10 @@
+#include "exported_funcs.h"
+
+namespace gandiva {
+REGISTER_EXPORTED_FUNCS(ExportedStubFunctions);
+REGISTER_EXPORTED_FUNCS(ExportedContextFunctions);
+REGISTER_EXPORTED_FUNCS(ExportedTimeFunctions);
+REGISTER_EXPORTED_FUNCS(ExportedDecimalFunctions);
+REGISTER_EXPORTED_FUNCS(ExportedStringFunctions);
+REGISTER_EXPORTED_FUNCS(ExportedHashFunctions);
```

Comment:
All the `REGISTER_EXPORTED_FUNCS` are moved into a `exported_funcs.cc` file, which makes the registration API to be called from 7 * 6 times down to 6 times.